src/stateProviders/provideLoggedInUser.js

```diff
@@ -48,11 +49,16 @@ export default provideState({
     setUser: (effects, { api, ...user }) => {
       return getAllFieldNamesPromise(api)
         .then(({ data }) => {
-          return get(data, '__type.fields', []).length;
+          return get(data, '__type.fields', []);
         })
-        .then(totalFields => state => {
-          const filledFields = Object.values(user || {}).filter(v => v || (isArray(v) && v.length));
-          const percentageFilled = filledFields.length / totalFields;
+        .then(fields => state => {
+          const userRole = user.roles ? user.roles[0] : null;
+          const userRoleProfileFields = userRole ?  find(ROLES, { type: userRole }).profileFields : fields;
+          const profile = pick(user, userRoleProfileFields);
+          const filledFields = Object.values(profile || {}).filter(
+            v => v || (isArray(v) && v.length),
```

Comment:
not sure if I understand this filter, afaik we'll never hit the fallback because an empty array is truthy